src/preferences.html

```diff
@@ -45,7 +45,7 @@
             </div>
             <div class="flex-box">
                 <p data-i18n="$Preferences.hoursPerDay">Hours per day</p>
-                <input data-i18n="[placeholder]$Preferences.hours-per-day" type="text" name="hours-per-day" id="hours-per-day" maxlength=5 pattern="^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$" value="08:00" size=5 required oninput="this.reportValidity()" onblur="this.value = this.checkValidity() ? this.value : '08:00'">
+                <input data-i18n="[placeholder]$Preferences.hours-per-day" type="text" name="hours-per-day" id="hours-per-day" maxlength=5 pattern="^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$" value="08:00" size=5 required oninvalid="this.setCustomValidity('Please enter the time in valid 24-hour format from 00:00 to 23:59')" oninput="this.setCustomValidity('');this.reportValidity()" onblur="this.value = this.checkValidity() ? this.value : '08:00'">
```

Comment:
We want to avoid having hard coded strings in the app. The best approach is to do something like:

In all `translation.json` under `locales/*`, add:

```
"$Generic": {
  "hours-on-invalid": "Please enter the time in valid 24-hour format from 00:00 to 23:59')"
}
```

And use here as:

```suggestion
                <input data-i18n="[placeholder]$Preferences.hours-per-day" type="text" name="hours-per-day" id="hours-per-day" maxlength=5 pattern="^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$" value="08:00" size=5 required oninvalid="this.setCustomValidity($Generic.hours-on-invalid)" oninput="this.setCustomValidity('');this.reportValidity()" onblur="this.value = this.checkValidity() ? this.value : '08:00'">
```

I'm not 100% sure if this will be enough to correctly show the strings. :/
If not, we can do a little more research.

Reply:
The suggested change did not work in the html files. So, I tried other ways out on the similar lines. Added oninvalid attribute with the key $Generic.hours-on-invalid(whose value is setCustomValidity(<msg>)) to data-i18n attribute in html files